hooks/post-process.js

```diff
@@ -58,67 +31,104 @@ module.exports = {
-
-          if (javaName !== schemaName) {
-            const newPath = path.resolve(newSourceDir, `${desiredClassName}.java`);
-            fs.renameSync(oldPath, newPath);
-            debugPostProcess(`Renamed class file ${schemaName} to ${desiredClassName}`);
-          }
-        } else {
-          // In this case it's an anonymous schema for a primitive type or something.
-          debugPostProcess(`deleting ${oldPath}`);
-          fs.unlinkSync(oldPath);
-        }
-      }
+function getDefaultJavaPackageDir(generator, defaultJavaPackage) {
+  let defaultPackageDir;
+
+  if (defaultJavaPackage) {
+    const packageDir = packageToPath(defaultJavaPackage);
+    defaultPackageDir = `${generator.targetDir}${sourceHead}${packageDir}`;
+  }
+
+  debugPostProcess(`getDefaultJavaPackageDir: ${defaultPackageDir}`);
+  return defaultPackageDir;
+}
+
+function packageToPath(javaPackage) {
+  return javaPackage.replace(/\./g, '/');
+}
+
+function processSchema(generator, schemaName, schema, sourcePath, defaultJavaPackageDir) {
+  const fileName = getFileName(schemaName);
+  const filePath = path.resolve(sourcePath, fileName);
+  debugPostProcess(`processSchema ${schemaName}`);
+  debugPostProcess(schema);
+  if (schema.type() !== 'object') {
+    debugPostProcess(`deleting ${filePath}`);
+    fs.unlinkSync(filePath);
+  } else {
+    const modelClass = applicationModel.getModelClass(schemaName);
+    const javaName = modelClass.getClassName();
+    const packageDir = getPackageDir(generator, defaultJavaPackageDir, modelClass);
+    debugPostProcess(`packageDir: ${packageDir}`);
+
+    if (packageDir) {
+      moveFile(sourcePath, packageDir, fileName);
     }
 
-    function moveFile(oldDirectory, newDirectory, fileName) {
-      if (!fs.existsSync(newDirectory)) {
-        fs.mkdirSync(newDirectory, { recursive: true });
-        debugPostProcess(`Made directory ${newDirectory}`);
-      }
-      const oldPath = path.resolve(oldDirectory, fileName);
-      const newPath = path.resolve(newDirectory, fileName);
-      fs.copyFileSync(oldPath, newPath);
-      fs.unlinkSync(oldPath);
-      debugPostProcess(`Moved ${fileName} from ${oldPath} to ${newPath}`);
+    debugPostProcess(`javaName: ${javaName} schemaName: ${schemaName}`);
+    if (javaName !== schemaName) {
+      const currentPath = packageDir || sourcePath;
+      const newPath = path.resolve(currentPath, `${javaName}.java`);
+      const oldPath = path.resolve(currentPath, fileName);
+      fs.renameSync(oldPath, newPath);
+      debugPostProcess(`Renamed class file ${schemaName} to ${javaName}`);
     }
   }
-};
+}
+
+function getFileName(schemaName) {
+  const trimmedSchemaName = trimSchemaName(schemaName);
+  return `${trimmedSchemaName}.java`;
+}
+
+function trimSchemaName(schemaName) {
+  let trimmedSchemaName = schemaName;
+  if (schemaName.startsWith('<')) {
+    trimmedSchemaName = schemaName.replace('<', '');
+    trimmedSchemaName = trimmedSchemaName.replace('>', '');
```

Comment:
It looks like it may be possible for a schema name to have angle brackets as part of the name according to the playground. Pretty sure replace starts from the left, so maybe the second replace would need to use a regex with `$` so `('/>$/', '')` with it properly escaped if needed

Reply:
Good catch, I'll definitely improve that.